smdebug/tensorflow/base_hook.py

```diff
@@ -369,13 +373,18 @@ def _log_unsupported_optimizer(self, optimizer):
 
     def _get_collections_with_tensor(self, tf_tensor_name) -> Set["Collection"]:
         self._assert_prep()
+        if self.tape:
+            return super()._get_collections_with_tensor(tf_tensor_name)
```

Comment:
why is this special one and why can't we put tf_tensor_name in tensor_to_collections map ?

Reply:
in general for TF, tensor_to_collections is populated in prepare_layers().

this is not applicable to GradientTape as the tensors are EagerTensors and can be written as soon as we get them through tape.gradient().

so in case of GradientTape, tensor_to_collections has to be populated through core/hook.py's _get_collections_with_tensor and there is a return of self.tensor_to_collections[tensor_name] there